intake_esm/core.py

```diff
@@ -483,6 +485,38 @@ def _allnan_or_nonan(column):
         self._ds = {group_id: ds for (group_id, ds) in dsets}
         return self._ds
 
+    def validate(self, esmcol_spec_dirs=None, version='master'):
+        """Validate ESM collection JSON file against the esm-collection-spec
+
+        Parameters
+        ----------
+        esmcol_spec_dirs: list, defaults to `None`
+          List of local specification directories to check for JSON schema files.
+        version : str, defaults to `master`
+            ESM Collection Specification version to validate against.
+            Uses github tags from the esm-collection-spec repository (https://github.com/NCAR/esm-collection-spec)
+            e.g.: v0.1.0
+
+        Returns
+        -------
+        status: dict
+           Dictionary of validation status
+        message: list
+           list of dictionaries containing validation messages
+
+        """
+        try:
+            from esmcol_validator.validator import EsmcolValidate
+        except ImportError:
+            raise ImportError(
+                """Using validate() method requires the `esmcol-validator` package. \nYou can install it via PyPI or Conda"""
+            )
+
+        log_level = logging.getLevelName(self._numeric_log_level)
+        esmcol = EsmcolValidate(self.esmcol_path, esmcol_spec_dirs, version, log_level)
+        esmcol.run()
+        return esmcol.status, esmcol.message
```

Comment:
Is there a reason why you want to return a `(status, message)` tuple instead of the `esmcol` object, itself?